stl/inc/random

```diff
@@ -261,30 +261,37 @@ private:
     vector<result_type> _Myvec;
 };
 
-_NODISCARD constexpr int _Generate_canonical_iterations(const int _Bits, const uint64_t _Gmin, const uint64_t _Gmax) {
-    // For a URBG type `G` with range == `(G::max() - G::min()) + 1`, returns the number of calls to generate at least
-    // _Bits bits of entropy. Specifically, max(1, ceil(_Bits / log2(range))).
-
-    _STL_INTERNAL_CHECK(0 <= _Bits && _Bits <= 64);
+struct _Urbg_gen_can_params {
+    int _Kx          = 0; // $k$ in [rand.util.canonical]/1.4
+    uint64_t _Xx     = 0; // $x$ in [rand.util.canonical]/1.4
+    int _Smax_bits   = 0; // number of bits in $R^k$
+    float _Scale     = 0.0f; // $r^{-d}$, always representable as a float
+    bool _Rx_is_pow2 = false;
+};
 
-    if (_Bits == 0 || (_Gmax == UINT64_MAX && _Gmin == 0)) {
-        return 1;
+_NODISCARD constexpr _Urbg_gen_can_params _Generate_canonical_params(size_t _Bits, uint64_t _Rm1) {

```

Comment:
This function is not supposed to be called at run time. Is it valuable to make it `consteval` since C++20?
But conditional compilation is possibly undesired as #3935 removed the internal macro `_CONSTEVAL`.

Reply:
Good question. As this is purely internal and used in only one place, I don't think conditionally using `consteval` is worth the effort.